alignak/scheduler.py

```diff
@@ -195,8 +194,10 @@ def reset(self):
         :return: None
         """
         self.must_run = True
-        with self.waiting_results_lock:
-            del self.waiting_results[:]
+
+        queue_size = self.waiting_results.qsize()
+        for _ in xrange(queue_size):
+            _ = self.waiting_results.get()
```

Comment:
Have a look at this to empty queue : http://stackoverflow.com/questions/6517953/clear-all-items-from-the-queue